posit-dev/positron
# Fix bokeh after holoviews

Addresses #4541 

This PR  does two main things:
1. Generalizes the previously-named`positronHoloVizService` to `positronWebviewPreloadService` and adds support for remembering bokeh loading messages. 
2. Patches the data-sending function that bokeh uses when in notebook mode to append a mime-type that lets the webview preload service know what to cache. 

It also re-enables the previously disabled bokeh smoke test. 


### QA Notes
Running the following code in order should now actually display both plots

```python
import hvplot.pandas
import pandas as pd
pd.DataFrame(dict(x=[1,2,3], y=[4,5,6])).hvplot.scatter(x="x", y="y")


from bokeh.plotting import figure, show
p = figure(title="Simple line example", x_axis_label='x', y_axis_label='y')
p.line([1, 2, 3, 4, 5], [6, 7, 2, 4, 5], legend_label="Temp.", line_width=2)
show(p)
```

Notebook mode for bokeh should also work.

```python
from bokeh.plotting import figure, show, output_notebook
output_notebook()
p = figure(title="Simple line example", x_axis_label='x', y_axis_label='y')
p.line([1, 2, 3, 4, 5], [6, 7, 2, 4, 5], legend_label="Temp.", line_width=2)
show(p)
```
